### ISSUE TITLE
Add pa11y accessibility testing to all patterns.

### ISSUE THREAD

**Issue by jcar3488** [phase2/particle#114]
https://github.com/pa11y/pa11y

Internal notes on potential platform wide approach: https://docs.google.com/document/d/1gXfdl1axMTbe8sNC2z5v3IFn-BiXEIHJaAnVptzkvKg/edit#heading=h.4asc00umu61p

@grayside, what is your opinion about letting us bring tooling into this design system instead of a container approach?

**Comment by illepic** [comment]
PR open here: https://github.com/phase2/pattern-lab-starter/pull/135

**Comment by grayside** [comment]
Well, since you pinged in that way, I'll share medium-sized thoughts about containers, backend tooling, and the questions this approach brings to mind.

I emphasized the significance of a container-based approach because of my general feeling that anything which requires a headless browser seems to involve large file sizes and/or special installation requirements. Purely on the speed of build process, for such cases using a container can be helpful as that piece of the testing can be entirely pre-built.

The reason I aimed for this more from a whole-project standpoint is that we need to have a facility to test accessibility not just for the design system, but also in a rubber-hits-the-editorial-road website.

I want to make sure that all testing gets folded in the CI process of projects that use PLS, currently we emphasize installation, code validation, and "compilation" as steps which allow the theme specific ways to participate in that wholistic/backend perspective of running processes.

From my perspective, here are the questions to dig into a bit:

* Can we find a good way to trigger this kind of testing from grunt-drupal-tasks et al? (I have ideas)
* Can this testing target a Drupal site or just the design system? Can it be flexible enough for local and alternate environments?
* Are there persnickety requirements for pa11y which would make merging this with other node dependencies tricky if it's not tested in that way on an ongoing basis?
* Does it impede the build process performance?
